librz/analysis/var_global.c

```diff
@@ -0,0 +1,423 @@
+// SPDX-FileCopyrightText: 2021 Basstorm <basstorm@nyist.edu.cn>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_analysis.h>
+#include <rz_list.h>
+
+/**
+ * \brief Create a new instance of global variable
+ * 
+ * \param name variable name
+ * \param addr variable address
+ * \param comment variable comment
+ * \return RzAnalysisVarGlobal *
+ */
+RZ_API RZ_OWN RzAnalysisVarGlobal *rz_analysis_var_global_new(RZ_NONNULL char *name, ut64 addr, char *comment) {
+	rz_return_val_if_fail(name && addr, NULL);
+	RzAnalysisVarGlobal *glob = RZ_NEW0(RzAnalysisVarGlobal);
+	if (!glob) {
+		return NULL;
+	}
+	glob->name = strdup(name);
+	glob->addr = addr;
+	if (comment) {
+		glob->comment = strdup(comment);
```

Comment:
`strdup()` on NULL will return an error. I recommend to either declare it `RZ_NONNULL` or just handle this case with `?:` operator. Also, if you use `strdup()` I recommend to define the argument as `const char*` instead.

Reply:
Judging by the function logic it should be `RZ_NULLABLE const char *comment`